lib/Fhp/PaginateableAction.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Fhp;
+
+use Fhp\Protocol\BPD;
+use Fhp\Protocol\Message;
+use Fhp\Protocol\UnexpectedResponseException;
+use Fhp\Protocol\UPD;
+use Fhp\Segment\HIRMS\Rueckmeldungscode;
+use Fhp\Segment\Paginateable;
+
+abstract class PaginateableAction extends BaseAction
+{
+    /**
+     * @var Stores the request created by BaseAction::getNextRequest to be reused in cases where the bank response
+     * can be split over multiple pages e.g. request/response pairs. This avoids the need for {@link BPD} to be
+     * available for paginated requests.
+     */
+    private $requestSegments;
+
+    /**
+     * If set, the last response from the server regarding this action indicated that there are more results to be
+     * fetched using this pagination token. This is called "Aufsetzpunkt" in the specification.
+     * @var string|null
+     */
+    private $paginationToken;
+
+    /** {@inheritdoc} */
+    public function serialize(): string
+    {
+        if (!$this->needsTan()) {
+            throw new \RuntimeException('Cannot serialize this action, because it is not waiting for a TAN.');
+        }
+        return serialize([$this->requestSegmentNumbers, $this->tanRequest, $this->needTanForSegment, $this->paginationToken, $this->requestSegments]);
```

Comment:
Geht auch `serialize([super::serialize(), $this->paginationToken, ..])`?

Dann müssten die Felder in der Super-Klasse nicht protected sein und man kann nichts falsch machen, wenn man in der Super-Klasse Felder hinzufügt/löscht.